### PR TITLE
ENT-11099: Update Java compile source & target to 17

### PR DESCRIPTION
**Pull request by shamsasari**
And removed unused `jdkClassifier` in build files.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -369,7 +362,7 @@ allprojects {
     if (jdkClassifier) {
```
Comment (arshadm) [inline build.gradle]: We can probably remove this, as we don't use the jdkClasssifier anymore in the project.

File: buildSrc/build.gradle [hunk 2]
```diff
@@ -53,3 +53,7 @@ dependencies {
         implementation "com.r3.internal.gradle.plugins:publish:$internalPublishVersion"
     }
 }
+
+jar {
+    duplicatesStrategy(DuplicatesStrategy.EXCLUDE)
+}
```
Comment (arshadm) [inline buildSrc/build.gradle]: Why do we need this in the buildSrc? I did add this to some build files, but there is a JIRA ticket to check where a duplicate dependency is coming from and fix the underlying issue. Most like a dodgy FatJar dependency.

Comment (shamsasari) [reply buildSrc/build.gradle]: The duplicate strategy thing is something they added in Gradle to force users to decide how to deal with duplicate files when zipping/jarring. I think previously it just ignored the duplicate. So the need to specify a strategy is not a issue.

As to why the buildSrc now needs it, not sure. I'm seeing this on all branches based off release/os/4.12.

**Comment by shamsasari** [comment]
PTAL

File: buildSrc/build.gradle (hunk 2 above)
Comment (shamsasari) [reply buildSrc/build.gradle]: Reverted, seems to be a local issue. And it's gone away for me.

**Comment by arshadm** [comment]
tools/explorer/build.gradle has a hardwired sourceCompatibilty of 1.8, not sure if that's really required.

**Review by rick-r3 — APPROVED** [review]
No comments from me

**Pull request merged**